Peer.Domain/State.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace Peer.Domain
+{
+    public class State
+    {
+        public PullRequestStatus Status { get; }
+        public int TotalComments { get; }
+        public int ActiveComments { get; }
+
+        public State(PullRequestStatus status, int totalComments, int activeComments)
+        {
+            if (!Enum.IsDefined(status))
```

Comment:
Use `Validators`?

Reply:
Fair, ported that over 3/4 the way through impl